diff --git a/var/spack/repos/builtin/packages/trilinos/package.py b/var/spack/repos/builtin/packages/trilinos/package.py
index d346b5e0a1..3292a8af1c 100644
--- a/var/spack/repos/builtin/packages/trilinos/package.py
+++ b/var/spack/repos/builtin/packages/trilinos/package.py
@@ -414,11 +414,11 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
         depends_on("kokkos-kernels~shared", when="+cuda_rdc")
         depends_on("kokkos-kernels~shared", when="+rocm_rdc")
         depends_on("kokkos~complex_align")
-        depends_on("kokkos@4.5.01", when="@master:")
+        depends_on("kokkos@4.6:", when="@master:")
         depends_on("kokkos@4.3.01", when="@16")
         depends_on("kokkos@4.2.01", when="@15.1:15")
         depends_on("kokkos@4.1.00", when="@14.4:15.0")
-        depends_on("kokkos-kernels@4.5.01", when="@master:")
+        depends_on("kokkos-kernels@4.6:", when="@master:")
         depends_on("kokkos-kernels@4.3.01", when="@16")
         depends_on("kokkos-kernels@4.2.01", when="@15.1:15")
         depends_on("kokkos+openmp", when="+openmp")
diff --git a/packages/stk/stk_tools/stk_tools/mesh_tools/EntityDisconnectTool.cpp b/packages/stk/stk_tools/stk_tools/mesh_tools/EntityDisconnectTool.cpp
index 5538709b240..3dec0b44992 100644
--- a/packages/stk/stk_tools/stk_tools/mesh_tools/EntityDisconnectTool.cpp
+++ b/packages/stk/stk_tools/stk_tools/mesh_tools/EntityDisconnectTool.cpp
@@ -97,8 +97,14 @@ bool EntityDisconnectTool::is_element_connected(const stk::mesh::Entity& face) c
 void EntityDisconnectTool::identify_adjacent_elements()
 {
   m_adjacentElements.clear();
-  m_adjacentElements.reserve(m_adjacentFaces.size() * 2U);
-  for (const auto& dcFace : m_adjacentFaces) {
+  m_adjacentElements.reserve(m_adjacentFaces.size() * 2U + m_disconnectFaces.size() * 2U);
+  for (const auto& adjFace : m_adjacentFaces) {
+    const auto dcElems = mesh().get_connected_entities(adjFace, stk::topology::ELEMENT_RANK);
+    for (auto e = 0U; e < dcElems.size(); ++e) {
+      m_adjacentElements.push_back(dcElems[e]);
+    }
+  }
+  for (const auto& dcFace : m_disconnectFaces) {
     const auto dcElems = mesh().get_connected_entities(dcFace, stk::topology::ELEMENT_RANK);
     for (auto e = 0U; e < dcElems.size(); ++e) {
       m_adjacentElements.push_back(dcElems[e]);