diff --git a/var/spack/repos/builtin/packages/trilinos/package.py b/var/spack/repos/builtin/packages/trilinos/package.py
index d346b5e0a1..3292a8af1c 100644
--- a/var/spack/repos/builtin/packages/trilinos/package.py
+++ b/var/spack/repos/builtin/packages/trilinos/package.py
@@ -414,11 +414,11 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
         depends_on("kokkos-kernels~shared", when="+cuda_rdc")
         depends_on("kokkos-kernels~shared", when="+rocm_rdc")
         depends_on("kokkos~complex_align")
-        depends_on("kokkos@4.5.01", when="@master:")
+        depends_on("kokkos@4.6.02", when="@master:")
         depends_on("kokkos@4.3.01", when="@16")
         depends_on("kokkos@4.2.01", when="@15.1:15")
         depends_on("kokkos@4.1.00", when="@14.4:15.0")
-        depends_on("kokkos-kernels@4.5.01", when="@master:")
+        depends_on("kokkos-kernels@4.6.02", when="@master:")
         depends_on("kokkos-kernels@4.3.01", when="@16")
         depends_on("kokkos-kernels@4.2.01", when="@15.1:15")
         depends_on("kokkos+openmp", when="+openmp")
